src/test/java/org/opensearch/neuralsearch/plugin/query/NeuralQueryBuilderTests.java

```diff
@@ -0,0 +1,326 @@
+            .field(MODEL_ID_FIELD.getPreferredName(), MODEL_ID)
+            .field(K_FIELD.getPreferredName(), K)
+            .field(QUERY_TEXT_FIELD.getPreferredName(), QUERY_TEXT)
+            .field(MODEL_ID_FIELD.getPreferredName(), MODEL_ID)
+            .field(K_FIELD.getPreferredName(), K)
+            .endObject()
+            .endObject();
+
+        XContentParser contentParser = createParser(xContentBuilder);
+        expectThrows(IOException.class, () -> NeuralQueryBuilder.fromXContent(contentParser));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testToXContent() throws IOException {
+        NeuralQueryBuilder neuralQueryBuilder = new NeuralQueryBuilder().fieldName(FIELD_NAME).modelId(MODEL_ID).queryText(QUERY_TEXT).k(K);
+
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        builder = neuralQueryBuilder.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        Map<String, Object> out = xContentBuilderToMap(builder);
+
+        Object outer = out.get(NAME);
+        if (!(outer instanceof Map)) {
+            fail("neural does not map to nested object");
+        }
+
+        Map<String, Object> outerMap = (Map<String, Object>) outer;
+
+        assertEquals(1, outerMap.size());
+        assertTrue(outerMap.containsKey(FIELD_NAME));
+
+        Object secondInner = outerMap.get(FIELD_NAME);
+        if (!(secondInner instanceof Map)) {
+            fail("field name does not map to nested object");
+        }
+
+        Map<String, Object> secondInnerMap = (Map<String, Object>) secondInner;
+
+        assertEquals(MODEL_ID, secondInnerMap.get(MODEL_ID_FIELD.getPreferredName()));
+        assertEquals(QUERY_TEXT, secondInnerMap.get(QUERY_TEXT_FIELD.getPreferredName()));
+        assertEquals(K, secondInnerMap.get(K_FIELD.getPreferredName()));
+    }
+
+    public void testStreams() throws IOException {
+        NeuralQueryBuilder original = new NeuralQueryBuilder();
+        original.fieldName(FIELD_NAME);
+        original.queryText(QUERY_TEXT);
+        original.modelId(MODEL_ID);
+        original.k(K);
+        original.boost(BOOST);
+        original.queryName(QUERY_NAME);
+
+        BytesStreamOutput streamOutput = new BytesStreamOutput();
+        original.writeTo(streamOutput);
+
+        NeuralQueryBuilder copy = new NeuralQueryBuilder(streamOutput.bytes().streamInput());
+        assertEquals(original, copy);
+    }
+
+    public void testHashAndEquals() {
+        String fieldName1 = "field 1";
+        String fieldName2 = "field 2";
+        String queryText1 = "query text 1";
+        String queryText2 = "query text 2";
+        String modelId1 = "model-1";
+        String modelId2 = "model-2";
+        float boost1 = 1.8f;
+        float boost2 = 3.8f;
+        String queryName1 = "query-1";
+        String queryName2 = "query-2";
+        int k1 = 1;
+        int k2 = 2;
+
+        NeuralQueryBuilder neuralQueryBuilder1 = new NeuralQueryBuilder().fieldName(fieldName1)
```

Comment:
rather than using 1,2,3,4,5 can we rename them in fashion with which just by variable name we can tell what is missing/what default values we have used

Reply:
Will update